ocean_lib/models/dtfactory.py

```diff
@@ -19,21 +19,24 @@ class DTFactory(ContractBase):
     def verify_data_token(self, dt_address):
         """Checks that a token was registered."""
         filter_params = {"tokenAddress": dt_address}
-        event_filter = self.events.TokenRegistered().createFilter(
-            fromBlock=0, argument_filters=filter_params
+        logs = ContractBase.getLogs(
+            self.events.TokenRegistered(),
+            argument_filters=filter_params,
+            fromBlock=0,
+            toBlock=999,
```

Comment:
This code won't work.  It only searches from blocks 0 to 999 and then gives up so TokenRegistered events from later blocks will never be found.

I believe this change was implemented because @mariacarmina ran into an [error while testing on Polygon](https://github.com/oceanprotocol/ocean.py/pull/309#issuecomment-880792556): "Blockheight too far in the past: eth_getLogs. Range of blocks allowed for your plan: 1000"

To accommodate this restriction, we must call `getLogs` in a loop, searching 1000 block chunks, starting at the latest block, and moving backward until the TokenRegistered event is found.

@mariacarmina when you implement the logic I've described, please use the `get_token_registered_event` instead of calling `ContractBase.getLogs` directly.

Reply:
Can we use `get_event_logs` from [here](https://github.com/oceanprotocol/ocean.py/blob/main/ocean_lib/web3_internal/contract_base.py#L271) in `get_token_registered_event`?

Reply:
You could but that wouldn't eliminate the need for a loop. My suggestion to use get_token_registered_event was merely to avoid duplicate calls.

Reply:
Upon closer inspection, I realize that get_event_logs contains similar loop logic to what I'm describing. So just use that.

Reply:
Exactly, that's why I have purposed it 👍🏻